### PR TITLE
SPRK-116 preapprovals example

### PR DESCRIPTION
**Pull request by nderkim**
## Description

Add form validation

## Associated Tickets

- [SPRK-116](https://brighte.atlassian.net/browse/SPRK-116)

### CODE REVIEW

File: examples/preapprovals/components/segmented-control.tsx [hunk 1]
```diff
@@ -1,66 +1,84 @@
...
-            })}
-          >
-            <Text>{option.label}</Text>
-          </Box>
-        </Box>
-      ))}
+          },
+          'input:active + &': {
+            background: theme.backgroundInteractions.primaryLowActive,
+          },
```
Comment (lukebennett88) [inline examples/preapprovals/components/segmented-control.tsx:77]: I just realised we should probably add focus styles.

Something like this should work:

```
// Import the focus styles from '@spark-web/a11y'
const focusStyles = useFocusRing();
```

And apply them when focused (along with position: relative and a z-index so the focus ring will be above the sibling radio inputs:

```suggestion
          'input:active + &': {
            background: theme.backgroundInteractions.primaryLowActive,
          },
          'input:focus + &': {
            ...focusStyles,
            position: 'relative',
            zIndex: theme.elevation.sticky,
          },
```

File: examples/preapprovals/components/step-2-form.tsx [hunk 2]
```diff
@@ -92,3 +310,28 @@ const Section = ({
...
 );
+
+const RecurringPaymentFrequencySelect = forwardRef<
+  HTMLSelectElement,
+  { options: { label: string; value: string }[] }
+>(function RecurringPaymentFrequencySelect({ options }, ref) {
+  const theme = useTheme();
+
+  return (
+    <select
```
Comment (lukebennett88) [inline examples/preapprovals/components/step-2-form.tsx:321]: We might need to add some extra style so the select looks good on Firefox too:

PixelSnap 2022-09-30 at 08 59 47@2x

File: examples/preapprovals/components/step-2-form.tsx [hunk 3]
```diff
@@ -1,67 +1,285 @@
...
       </Section>
       <Section heading="Income">
-        <Field label="Base net income">
-          <RecurringPaymentInput />
+        <Field
+          label="Base net income"
+          {...(formFieldError(form, 'baseNetIncome') ||
+            formFieldError(form, 'baseNetIncomeFrequency'))}
+        >
+          <TextInput type="number" {...form.register('baseNetIncome')}>
```
Comment (lukebennett88) [inline examples/preapprovals/components/step-2-form.tsx:173]: I'd get rid of `type="number"` on these.

- https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/
- https://bradfrost.com/blog/post/you-probably-dont-need-input-typenumber/

Comment (nderkim) [reply examples/preapprovals/components/step-2-form.tsx]: yeh nice! i didn't like that it kept changing the values on scroll.

File: examples/preapprovals/components/segmented-control.tsx (hunk 1 above)
Comment (nderkim) [reply examples/preapprovals/components/segmented-control.tsx:78]: done ✅

File: examples/preapprovals/components/step-2-form.tsx (hunk 2 above)
Comment (nderkim) [reply examples/preapprovals/components/step-2-form.tsx:315]: i think i'll just come back to this later in another PR

**Review by lukebennett88 — APPROVED** [review]
🧐

**Pull request merged**